tests/integration/command_test.py

```diff
@@ -0,0 +1,241 @@
+"""Various tests for `candies.cli.command.Command`."""
+
+from typing import Any
+
+import pytest
+
+from candies.cli.args.arg import Arg
+from candies.cli.args.opt import Opt
+from candies.cli.args.flag import Flag
+from candies.cli.cli import cli, CLI
+
+
+def execute(with_: str) -> Any:
+    try:
+        return cli(with_.split())
+    # `SystemExit`, because `argparse` calls `exit` on error.
+    except SystemExit as e:
+        return e
+
+
+@pytest.fixture(autouse=True)
+def reset():
+    # Reset `cli` before each test
+    # as if we have just imported it.
+    globals()['cli'] = CLI()
+
+
+def test_valid_command_name():
+    # Arrange.
+    @cli
+    def main():
+        pass
+
+    @main.command
+    def one():
+        pass
+
+    @main.command
+    def two():
+        pass
+
+    @one.command
+    def a():
+        return True
+
+    @one.command
+    def b():
+        return False
+
+    @two.command
+    def x():
+        return True
+
+    @two.command
+    def y():
+        return False
+
+    # Act.
+    cx_a = execute(with_='one a')
+    cx_b = execute(with_='one b')
+    cx_x = execute(with_='two x')
+    cx_y = execute(with_='two y')
+
+    # Assert.
+    assert cx_a is True
+    assert cx_b is False
+    assert cx_x is True
+    assert cx_y is False
+
+
+def test_invalid_command_name():
```

Comment:
I'm not quite sure what does the `invalid_command_name` mean? Is it just a name of the command, or how it was specified in the command line?

Reply:
It's possible to reduce this test case to the following:
```py
# Arrange.
def test_one_subcommand_specified_incorrectly():
    @cli
    def main():
        pass

    @main.command
    def one():
        pass

    @one.command
    def two():
        pass

# Act.
ex = execute('one tw_')

# Assert.
...
```

And we also could have a mirror test case that checks happy path:
```py
# Arrange.
def test_one_subcommand():
    @cli
    def main():
        pass

    @main.command
    def one():
        pass

    @one.command
    def two():
        return True

# Act.
cx = execute('one two')

# Assert.
assert cx == True
```

Reply:
In this case, the name of an inexistent command is meant. Will rename the test to make it more clear.